src/Examples/UnitModels/Operations/statejunction_testing.ipynb

```diff
@@ -0,0 +1,464 @@
+    "\n",
+    "Mole Fraction (benzene) = 0.6\n",
+    "\n",
+    "Mole Fraction (toluene) = 0.4\n",
+    "\n",
+    "**Case 2**\n",
+    "\n",
+    "Volumetric Flow Rate = 10 m^3/s\n",
+    "\n",
+    "H2O Concentration = 5000 mol/m^3\n",
+    "\n",
+    "NaOH Concentration = 25 mol/m^3\n",
+    "\n",
+    "Temperature = 298.15 K\n",
+    "\n",
+    "Pressure = 101325 Pa \n",
+    "\n",
+    "We will look at two cases in this tutorial:\n",
+    "\n",
+    "* Case 1: Use the StateJunction unit model to solve a problem with constraints (requires a flowsheet-level solver).\n",
+    "\n",
+    "* Case 2: Use the StateJunction unit model to solve a problem without constraints (does not require a flowsheet-level solver).\n",
+    "\n",
+    "\n",
+    "For more details, please refer to the IDAES documentation: https://idaes-pse.readthedocs.io/en/stable"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting up the problem in IDAES"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the following cell, we will be importing the necessary components from Pyomo and IDAES."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import objects from pyomo package \n",
+    "from pyomo.environ import ConcreteModel, Constraint, value, SolverFactory\n",
+    "\n",
+    "# Import the solver\n",
+    "from idaes.core.solvers import get_solver\n",
+    "\n",
+    "# Import the main FlowsheetBlock from IDAES. The flowsheet block will contain the unit model\n",
+    "from idaes.core import FlowsheetBlock\n",
+    "\n",
+    "# Import the StateJunction unit model\n",
+    "from idaes.models.unit_models import StateJunction\n",
+    "\n",
+    "# Import idaes logger to set output levels\n",
+    "import idaes.logger as idaeslog\n",
+    "\n",
+    "# Import the modular property package to create a property block for the case 1 flowsheet\n",
+    "from idaes.models.properties.modular_properties.base.generic_property import GenericParameterBlock\n",
+    "\n",
+    "# Import the BT_ideal property package to create a configuration file for the GenericParameterBlock based on the ideal eos\n",
+    "from idaes.models.properties.modular_properties.examples.BT_ideal import configuration as configuration\n",
+    "\n",
+    "# Import the saponification property package to create a property block for the case 2 flowsheet\n",
+    "from idaes.models.properties.examples.saponification_thermo import SaponificationParameterBlock\n",
+    "\n",
+    "# Import the degrees_of_freedom function from the idaes.core.util.model_statistics package\n",
+    "# DOF = Number of Model Variables - Number of Model Constraints\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "\n",
+    "# Create the ConcreteModel and the FlowsheetBlock objects, and attach the flowsheet block to it.\n",
```

Comment:
It might help to break this into two parts - the imports and the actual flowsheet.